pkg/private/archive.py

```diff
@@ -174,11 +174,14 @@ def __init__(self,
                                               stdout=open(name, 'wb'))
       self.fileobj = self.compressor_proc.stdin
     self.name = name
-    self.root_directory = root_directory.rstrip('/').rstrip('\\')
-    self.root_directory = self.root_directory.replace('\\', '/')
+    # tarfile uses / instead of os.path.sep
+    self.root_directory = root_directory.replace(os.path.sep, '/').rstrip('/')
+    if self.root_directory:
+      self.root_directory = self.root_directory + '/'
```

Comment:
I don't want to rely on path manipulations because I want it explicit that we are
normalizing from the os sep to /.   We may be processing  paths of generated output files which can have \ in them on windows.

Reply:
IMO this shouldn't matter and would make the codebase a little more consistent.

For example, our UNIXy archive writer module could be written to assume that they receive UNIX-style paths, with the normalization to UNIX-style paths done earlier prior to invoking the archive writer.